bots/pr/src/main/java/org/openjdk/skara/bots/pr/ReviewCoverage.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.skara.bots.pr;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.openjdk.skara.forge.PullRequest;
+import org.openjdk.skara.forge.PullRequestUtils;
+import org.openjdk.skara.forge.Review;
+import org.openjdk.skara.vcs.Repository;
+import org.openjdk.skara.vcs.git.GitRepository;
+
+public class ReviewCoverage {
+
+    private final Logger log = Logger.getLogger("org.openjdk.skara.bots.pr");
+    private final boolean ignoreStaleReviews;
+    private final boolean includeSimpleMerges;
+    private final Repository repo;
+
+    public ReviewCoverage(boolean ignoreStaleReviews, boolean includeSimpleMerges, Repository repo) {
+        this.ignoreStaleReviews = ignoreStaleReviews;
+        this.includeSimpleMerges = includeSimpleMerges;
+        this.repo = repo;
+    }
+
+    public boolean covers(Review review, PullRequest pr) {
+        var r = review.hash();
+        // Reviews without a hash are never valid as they referred to no longer
+        // existing commits.
+        if (r.isEmpty() || review.verdict() != Review.Verdict.APPROVED
+                || !review.targetRef().equals(pr.targetRef())) {
+            return false;
+        }
+        if (!ignoreStaleReviews || r.get().equals(pr.headHash())) {
+            return true;
+        }
+        if (!includeSimpleMerges) {
+            return false;
+        }
```

Comment:
But here, if `includeSimpleMerges==true`, this method will try to ignore the simple merges.
if `includeSimpleMerges==false`, this method will not ignore the simple merges